lib/utilities/hash.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+module.exports = async function hash(obj) {
```

Comment:
Tests? Should we extract? If so, I can gladly do so today.

Reply:
@stefanpenner - Ya, I can write some tests for the util. I was wondering if we should think about a larger refactor in RSVP itself to make RSVP use the native promise if possible...

Reply:
@stefanpenner extracted into [promise.hash.helper](https://github.com/stefanpenner/promise.hash.helper)